### PR TITLE
Profiling

### PR DESCRIPTION
**Pull request by vlhtg**
I added the ability to track CPU and RAM usage. Don't bother with the implementation that much, but test it on your machine. I don't really understand the expected behaviour of the main branch stuff tbh ngl no cap.

Pls

### CODE REVIEW

**Comment by vlhtg** [comment]
```
   File "/home/runner/work/runner_container/runner_container/.github/processMetrics.py", line 113, in <module>
    main()
  File "/home/runner/work/runner_container/runner_container/.github/processMetrics.py", line 84, in main
    percentages_dependencies = compute_percentages_dependencies(matrix, dependency_columns, num_files)
  File "/home/runner/work/runner_container/runner_container/.github/processDependencies.py", line 28, in compute_percentages_dependencies
    if matrix[i][j] != 0 and matrix[j][i] != 0 and i != j:
IndexError: index 12 is out of bounds for axis 0 with size 12
```
Dude, this shit crashes. The checker I mean. :skull:

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 
-# install basic dependencies at creation time
+# add basic dependencies
+RUN apt-get update
+RUN apt-get install -y sysstat
+# this is for mpstat
+
 #RUN apt-get update -y && apt-get install -y python3 python3-pip python3-venv 
 
 WORKDIR /
```
Comment (TPGamesNL) [inline Dockerfile:10]: I wouldn't recommend using the root as the application directory, as there's a lot of other things there already. E.g. `/usr/local/bin/myname`

Comment (vlhtg) [reply Dockerfile:10]: broski, that was there before me.

File: profiler.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+while true
+do
+  # get CPU usage using mpstat
+  cpu_usage=$(mpstat 1 1 | awk '/all/ {print 100 - $12}' | head -n 1)
+  # this includes a second wait, while it calculates the average
+  
+  # get RAM usage using free
+  # ram_usage=$(free -m | awk 'NR==2{printf "%.2f", $3*100/$2}') # this is the percentage
+  ram_usage=$(free -m | awk 'NR==2{print $3 " MB"}') # this is the absolute value
```
Comment (TPGamesNL) [inline profiler.sh:20]: I think it'd be better to remove the `MB` part, then it's easier to interpret the data later, cause we can directly parse it as int (also its already implied by the header name 'RAM Usage MB')

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**